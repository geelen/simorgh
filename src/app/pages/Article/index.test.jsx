import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import ArticleContainer from './index';

// explicitly ignore console.log errors for Article/index:getInitialProps() error logging
global.console.log = jest.fn();

const defaultProps = {
  isAmp: false,
  pageType: 'article',
  service: 'news',
  pathname: '/pathname',
  status: 200,
};

jest.mock('../../containers/PageHandlers/withVariant', () => Component => {
  const VariantContainer = props => (
    <div id="VariantContainer">
      <Component {...props} />
    </div>
  );

  return VariantContainer;
});

jest.mock('../../containers/PageHandlers/withPageWrapper', () => Component => {
  const PageWrapperContainer = props => (
    <div id="PageWrapperContainer">
      <Component {...props} />
    </div>
  );

  return PageWrapperContainer;
});

jest.mock('../../containers/PageHandlers/withLoading', () => Component => {
  const LoadingContainer = props => (
    <div id="LoadingContainer">
      <Component {...props} />
    </div>
  );

  return LoadingContainer;
});

jest.mock('../../containers/PageHandlers/withError', () => Component => {
  const ErrorContainer = props => (
    <div id="ErrorContainer">
      <Component {...props} />
    </div>
  );

  return ErrorContainer;
});

jest.mock('../../containers/PageHandlers/withData', () => Component => {
  const DataContainer = props => (
    <div id="DataContainer">
      <Component {...props} />
    </div>
  );

  return DataContainer;
});

jest.mock('../../containers/ArticleMain', () => {
  const ArticleMain = () => <div>ArticleMain</div>;

  return ArticleMain;
});

describe('ArticleContainer', () => {
  describe('Component', () => {
    describe('Composing the Article Container using the page handlers', () => {
      shouldMatchSnapshot(
        'should compose articleContainer with the Page Handler in the correct order',
        <ArticleContainer {...defaultProps} />,
      );
    });
  });
});
