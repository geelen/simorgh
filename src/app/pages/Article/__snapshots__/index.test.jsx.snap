// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContainer Component Composing the Article Container using the page handlers should compose articleContainer with the Page Handler in the correct order 1`] = `
<div
  id="VariantContainer"
>
  <div
    id="PageWrapperContainer"
  >
    <div
      id="LoadingContainer"
    >
      <div
        id="ErrorContainer"
      >
        <div
          id="DataContainer"
        >
          <div>
            ArticleMain
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
